### PR TITLE
Verify emails and edit user profile for non-members

### PR DESCRIPTION
**Pull request by fredrikved**
PR implements gql endpoints for manipulating non-member accounts.

### CODE REVIEW

File: backend/auth0/management.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	token, err := client.GetToken(ctx, client.config.ManagementAudience)
+	if err != nil {
+		return
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+
+	res, err := http.DefaultClient.Do(req)
```
Comment (KillerX) [inline backend/auth0/management.go:37]: Default client has infinite timeout, maybe best not to use it.

File: backend/graph/api/schema/mutations.graphqls [hunk 2]
```diff
@@ -26,6 +26,9 @@ type MutationRoot {
     addShowToMyList(showId: ID!): AddToCollectionResult!
 
     removeEntryFromMyList(entryId: UUID!): UserCollection!
+
+    updateUserMetadata(birthData: BirthOptions!, nameData: NameOptions!): Boolean!
+    verifyEmail: Boolean!
```
Comment (KillerX) [inline backend/graph/api/schema/mutations.graphqls:31]: Is this actually `verify` or is it more `sendVerificationEmail`.

Comment (fredrikved) [reply backend/graph/api/schema/mutations.graphqls:31]: definitely sendVerificationEmail 🙃

File: backend/auth0/management.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		bodyReader = bytes.NewReader(stringBody)
+	}
+
+	managementUrl := fmt.Sprintf("https://%s/api/v2/", client.config.Domain)
+	req, err := http.NewRequestWithContext(ctx, method, fmt.Sprintf("%s%s", managementUrl, path), bodyReader)
+	if err != nil {
+		return
+	}
+
+	token, err := client.GetToken(ctx, client.config.ManagementAudience)
```
Comment (rvanoord) [inline backend/auth0/management.go:35]: In general I'm missing "scope" from the code that requests/caches tokens. I assume that means we are getting a token with "all" scopes which isn't really ideal.

File: backend/graph/api/mutations.resolvers.go [hunk 4]
```diff
@@ -451,6 +453,48 @@ func (r *mutationRootResolver) RemoveEntryFromMyList(ctx context.Context, entryI
...
+// UpdateUserMetadata is the resolver for the updateUserMetadata field.
+func (r *mutationRootResolver) UpdateUserMetadata(ctx context.Context, birthData model.BirthOptions, nameData model.NameOptions) (bool, error) {
+	ginCtx, err := utils.GinCtx(ctx)
+	if err != nil {
+		return false, err
+	}
+	u := user.GetFromCtx(ginCtx)
+	if u.IsRegistered() && !u.IsActiveBCC() {
+		displayName := nameData.First + " " + nameData.Last
+		_, err = r.AuthClient.UpdateUser(ctx, ginCtx.GetString(auth0.CtxUserID), auth0.UserInfo{
```
Comment (rvanoord) [inline backend/graph/api/mutations.resolvers.go:465]: Can we add some meta information here that identifies the user as a "BCC Media" subscriber?

File: backend/user/middleware.go [hunk 5]
```diff
@@ -215,13 +215,24 @@ func NewUserMiddleware(queries *sqlc.Queries, remoteCache *remotecache.Client, l
 					return i.OrgID
 				})
 			} else {
-				info, err := auth0Client.GetUserInfoForAuthHeader(ctx, ctx.GetHeader("Authorization"))
+				info, err := auth0Client.GetUser(ctx, ctx.GetString(auth0.CtxUserID))
 				if err != nil {
 					return nil, err
 				}
-				u.PersonID = info.Sub
+				u.PersonID = info.UserId
```
Comment (rvanoord) [inline backend/user/middleware.go:222]: Long term we probably shouldn't use the auth0 ID as the primary key. But that can work for now.

File: backend/auth0/management.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+type jobResponse struct {
+	Status    string
+	Type      string
+	CreatedAt string `json:"created_at"`
+	ID        string
+}
+
+// SendVerificationEmail sends a verification email to the specified user
+func (c *Client) SendVerificationEmail(ctx context.Context, sub string) error {
```
Comment (rvanoord) [inline backend/auth0/management.go:81]: To me it appears that SendVerificationEmail is initiated on the client side via GraphQL (not by the API) and that there isn't a filter in place to ensure it is only sent to BCC Media subscriber users

**Review by rvanoord — COMMENTED** [review]
Reviewed from a logic point of view (not code quality etc.)

File: backend/graph/api/mutations.resolvers.go (hunk 4 above)
Comment (fredrikved) [reply backend/graph/api/mutations.resolvers.go:465]: I thought that was going to be implemented by you guys or?

File: backend/auth0/management.go (hunk 3 above)
Comment (fredrikved) [reply backend/auth0/management.go:35]: If necessary. We're using all the scopes we have access to.

File: backend/user/middleware.go (hunk 5 above)
Comment (fredrikved) [reply backend/user/middleware.go:222]: Once we get another key to use, no problem implementing it. We aren't actually using this for much except that.

File: backend/auth0/management.go (hunk 6 above)
Comment (fredrikved) [reply backend/auth0/management.go:81]: Would that be necessary? Any user that doesn't have a verified email in Auth0 should be able to verify their email as well. Doesn't matter if it's created by the Media client or not.

**Review by KillerX — DISMISSED** [review]

File: backend/auth0/management.go (hunk 3 above)
Comment (rvanoord) [reply backend/auth0/management.go:35]: ok, if the client is only used within this app

File: backend/auth0/management.go (hunk 6 above)
Comment (rvanoord) [reply backend/auth0/management.go:81]: How does verification get triggered? is it automatic after first login, or is it something the user invokes?

**Pull request merged**